src/transaction.rs

```diff
@@ -76,7 +94,21 @@ impl CommitTransaction {
                  spending conditions using (X_0, R_0, Y_0) and (X_1, R_1, Y_1)"
                 ),
             }],
-        })
+        };
+
+        let digest = SighashComponents::new(&transaction).sighash_all(
+            &input,
+            // TODO: May need to instead call `.witness_script()` on the
+            // descriptor used to produce the `FundingTransaction`'s output
+            // `script_pubkey`
+            &input.script_sig,
+            TX_f.value().as_sat(),
+        );
```

Comment:
We might be able to not need to use SighashComponents. Checkout this example here: 

https://github.com/rust-bitcoin/rust-miniscript/blob/87a47e242dc19c517ca06c558cf79b43eec22abf/examples/verify_tx.rs#L116-L120